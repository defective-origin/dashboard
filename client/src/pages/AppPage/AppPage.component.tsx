import React from 'react'

// ---| core |---
import { useSystemLauncher, Router } from 'Launcher'

// ---| pages |---
import StatusPage from 'pages/StatusPage'
import HotKeysPage from 'pages/HotKeysPage'

// ---| screens |---

// ---| components |---
import Page, { PageProps } from 'components/Page'

// ---| common |---
import { cn } from 'common/tools'

// ---| self |---
import css from './AppPage.module.scss'
import AppMenuScreen from './AppMenuScreen'

export type AppPageProps = PageProps & {
  className?: string
}

/**
 * Component description.
 *
 * How to use
 * @example
 * <AppPage />
 */
export function AppPage(props: AppPageProps): JSX.Element {
  const { className, ...otherProps } = props
  const _className = cn(css.AppPage, className)
  const system = useSystemLauncher()

  return (
    <Page
      className={_className}
      name={system.t('SYSTEM.TAB_NAME', { title: system.t('PAGES.STATUS') })}
      type='left-aside'
      {...otherProps}
    >
      <Page.LeftAside>
        <AppMenuScreen />
      </Page.LeftAside>

      <Page.Content>
        <Router>
          <StatusPage path='/hotkeys' type='welcome' />
          {/* <DashboardPage path='/dashboard/:id' />
          <WidgetPage path='/widget/:id' />
          <AccountPage path='/account' />
          <GuidePage path='/guide' />
          <DonationPage path='/donation' /> */}
          <HotKeysPage path='/' />
          {/* <SupportPage path='/support' /> */}
          <StatusPage path='/error/:type' />
          <StatusPage default type={404} />
        </Router>
      </Page.Content>
    </Page>
  )
}

export default AppPage
