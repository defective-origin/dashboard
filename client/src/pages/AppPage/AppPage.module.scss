@import "src/Launcher/UILauncher/UILauncher.module.scss";

.AppPage {
}
