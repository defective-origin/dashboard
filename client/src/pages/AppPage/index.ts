export { default } from './AppPage.component'
export * from './AppPage.component'
