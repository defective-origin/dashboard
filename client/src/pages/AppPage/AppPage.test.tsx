// ---| tests |---
import { render } from 'tests'

// ---| self |---
import AppPage from './AppPage.component'

describe('[AppPage] component', () => {
  it('should render component', () => {
    const container = render(<AppPage />)

    expect(container.snapshot()).toMatchSnapshot()
  })
})
