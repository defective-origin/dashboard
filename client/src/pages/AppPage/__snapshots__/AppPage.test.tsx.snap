// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`[AppPage] component > should render component 1`] = `
<DocumentFragment>
  <div
    class="layout layout--left-aside page _AppPage_1f540f"
  >
    <div
      class="layout-item layout-item--left-aside"
    >
      <div
        class="block block--column-center _AppMenuScreen_b5cb46"
      >
        <div
          class="block-item block-item--start _TopActions_b5cb46"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-2s90m6-MuiAvatar-root"
          >
            <span
              aria-hidden="true"
              class="material-icons notranslate MuiIcon-root MuiIcon-fontSizeLarge _Icon_867a49 material-symbols-outlined css-1gb9piy-MuiIcon-root"
            >
              account_circle
            </span>
          </div>
        </div>
        <div
          class="block-item block-item--center _MiddleActions_b5cb46"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge _Button_7cc380 _SquareButton_b5cb46 css-mf1cb5-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <span
              aria-hidden="true"
              class="material-icons notranslate MuiIcon-root MuiIcon-fontSizeMedium _Icon_867a49 material-symbols-outlined css-kp9ftd-MuiIcon-root"
            >
              dashboard
            </span>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge _Button_7cc380 _SquareButton_b5cb46 css-mf1cb5-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <span
              aria-hidden="true"
              class="material-icons notranslate MuiIcon-root MuiIcon-fontSizeMedium _Icon_867a49 material-symbols-outlined css-kp9ftd-MuiIcon-root"
            >
              insert_chart
            </span>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="block-item block-item--end _BottomActions_b5cb46"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge _Button_7cc380 _SquareButton_b5cb46 _ActiveButton_b5cb46 css-mf1cb5-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <span
              aria-hidden="true"
              class="material-icons notranslate MuiIcon-root MuiIcon-fontSizeMedium _Icon_867a49 material-symbols-outlined css-kp9ftd-MuiIcon-root"
            >
              auto_stories
            </span>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge _Button_7cc380 _SquareButton_b5cb46 css-mf1cb5-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <span
              aria-hidden="true"
              class="material-icons notranslate MuiIcon-root MuiIcon-fontSizeMedium _Icon_867a49 material-symbols-outlined css-kp9ftd-MuiIcon-root"
            >
              paid
            </span>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge _Button_7cc380 _SquareButton_b5cb46 css-mf1cb5-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <span
              aria-hidden="true"
              class="material-icons notranslate MuiIcon-root MuiIcon-fontSizeMedium _Icon_867a49 material-symbols-outlined css-kp9ftd-MuiIcon-root"
            >
              keyboard
            </span>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge _Button_7cc380 _SquareButton_b5cb46 css-mf1cb5-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <span
              aria-hidden="true"
              class="material-icons notranslate MuiIcon-root MuiIcon-fontSizeMedium _Icon_867a49 material-symbols-outlined css-kp9ftd-MuiIcon-root"
            >
              support_agent
            </span>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge _Button_7cc380 _SquareButton_b5cb46 css-mf1cb5-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <span
              aria-hidden="true"
              class="material-icons notranslate MuiIcon-root MuiIcon-fontSizeMedium _Icon_867a49 material-symbols-outlined css-kp9ftd-MuiIcon-root"
            >
              light_mode
            </span>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall _Button_7cc380 _SquareButton_b5cb46 css-1knaqv7-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            en
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge _Button_7cc380 _SquareButton_b5cb46 css-mf1cb5-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <span
              aria-hidden="true"
              class="material-icons notranslate MuiIcon-root MuiIcon-fontSizeMedium _Icon_867a49 material-symbols-outlined css-kp9ftd-MuiIcon-root"
            >
              developer_mode_tv
            </span>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      class="layout-item layout-item--content"
    >
      <div
        class="_Router_318ffa"
        style="outline: none;"
        tabindex="-1"
      >
        <div
          class="layout layout--row page _HotKeysPage_a8be87"
          location="[object Object]"
          path="/"
          uri="/"
        >
          <div
            class="layout-item layout-item--content"
          >
            <div>
              TEST CONTENT
            </div>
            <div>
              TEST CONTENT
            </div>
            <div>
              TEST CONTENT
            </div>
            <div>
              TEST CONTENT
            </div>
            <div>
              TEST CONTENT
            </div>
            <div>
              TEST CONTENT
            </div>
          </div>
          <div
            class="layout-item layout-item--footer"
          >
            FOOTER
          </div>
          <div
            class="layout-item layout-item--left-aside"
          >
            LEFT ASIDE
          </div>
          <div
            class="layout-item layout-item--right-aside"
          >
            RIGHT ASIDE
          </div>
          <div
            class="layout-item layout-item--right-aside"
          >
            -------------------
          </div>
          <div
            class="layout-item layout-item--right-aside"
          >
            CONTENT
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
